### PR TITLE
chore: separate tslint config files for build and vscode

### PR DESCRIPTION
**Pull request by raymondfeng**
This is a follow-up to https://github.com/strongloop/loopback-next/pull/964/.

Two config files are in place now:
- tslint.build.json for CLI build scripts
- tslint.json for vscode

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

**Comment by raymondfeng** [comment]
> IIUC, this is an undo of the previous commit. But we will end up without a check for unused import again.

Not really. Please note that CLI build uses `tslint.build.json` and it catches violation of unused vars.

**Comment by virkt25** [comment]
By CLI do you mean `@loopback/build` and the `package.json` file?

**Comment by raymondfeng** [comment]
> By CLI do you mean @loopback/build and the package.json file?

`npm run lint` which uses `lb-tslint`.

**Comment by raymondfeng** [comment]
Yes, `lb-tslint` uses `tslint.build.json` over `tslint.json`. This way, `vscode` uses `tslint.json` while our build uses `tslint.build.json`.

File: packages/cli/generators/project/templates/tslint.build.json [hunk 1]
```diff
@@ -1,17 +1,4 @@
...
-    // These rules find errors related to TypeScript features.
-
-    // These rules catch common errors in JS programming or otherwise
-    // confusing constructs that are prone to producing bugs.
-
-    "await-promise": true,
-    "no-floating-promises": true,
-    "no-void-expression": [true, "ignore-arrow-function-shorthand"]
-  }
+  "extends": ["./node_modules/@loopback/build/config/tsconfig.build.json"]
```
Comment (bajtos) [inline packages/cli/generators/project/templates/tslint.build.json]: I think you have a typo here - are you sure that `tslint.build.json` should be extending `tsconfig.build.json`? I think you meant to extend `tslint.build.json` instead.

Comment (raymondfeng) [reply packages/cli/generators/project/templates/tslint.build.json]: Good catch. Fixed.

**Pull request merged**